package dmGameSystemDDF;

import "ddf/ddf_extensions.proto";
import "ddf/ddf_math.proto";

option java_package = "com.dynamo.sprite.proto";
option java_outer_classname = "Sprite";
option (ddf_java_package) = "com.dynamo.sprite.ddf";

message SpriteDesc
{
    required string Texture = 1 [(resource)=true];
    optional uint32 TileWidth = 2 [default = 0];
    optional uint32 TileHeight = 3 [default = 0];
    optional uint32 RowCount = 4 [default = 1];
    optional uint32 ColumnCount = 5 [default = 1];
}

message PlayAnimation
{
    required uint32 StartFrame = 1;
    required uint32 EndFrame = 2;
    required uint32 Loop = 3;
    required float FPS = 4;
}

message AnimationDone
{
    required uint32 CurrentFrame = 1;
}
